bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,241 @@
+        this.qbusComm.setBridgeCallBack(this);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+    private void setupRefreshTimer(Integer refreshInterval) {
+        if (this.refreshTimer != null) {
+            this.refreshTimer.cancel(true);
+            this.refreshTimer = null;
+        }
+
+        if ((refreshInterval == null) || (refreshInterval == 0)) {
+            return;
+        }
+
+        // This timer will restart the bridge connection periodically
+        logger.debug("Qbus: Checking for Client communication every {} min", refreshInterval);
+        this.refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            logger.debug("Qbus: check communication after timerinterval");
+
+            if (!qbusComm.communicationActive()) {
+                logger.debug("Qbus: Restarting communication");
+                qbusComm.restartCommunication();
+
+                if (!qbusComm.communicationActive()) {
+                    qbusComm = null;
+                    bridgeOffline();
+                    updateStatus(ThingStatus.OFFLINE);
+                    return;
+                }
+
+                // updateStatus(ThingStatus.ONLINE);
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.debug("Qbus: Communication still active");
+            }
+
+        }, refreshInterval, refreshInterval, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Take bridge offline when error in communication with Qbus server. This method can also be
+     * called directly from {@link QbusCommunication} object.
+     */
+    public void bridgeOffline() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                "Qbus: error starting bridge connection");
+    }
+
+    /**
+     * Put bridge online when error in communication resolved.
+     */
+    public void bridgeOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public boolean isInitialized() {
+        return true;
+    }
+
+    @Override
+    public void dispose() {
+        if (this.refreshTimer != null) {
+            this.refreshTimer.cancel(true);
+        }
+        this.refreshTimer = null;
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        Configuration configuration = editConfiguration();
+        for (Entry<String, Object> configurationParmeter : configurationParameters.entrySet()) {
+            configuration.put(configurationParmeter.getKey(), configurationParmeter.getValue());
+        }
+        updateConfiguration(configuration);
+
+        scheduler.submit(() -> {
+
+            updateStatus(ThingStatus.ONLINE);
+
+            Integer refreshInterval = ((Number) configuration.get(CONFIG_REFRESH)).intValue();
+            setupRefreshTimer(refreshInterval);
+        });
+    }
```

Comment:
Is this really necessary? `initialize()` is called when the configuration has changed.